exonum/src/crypto.rs

```diff
@@ -129,6 +131,17 @@ pub fn hash(data: &[u8]) -> Hash {
     Hash(dig)
 }
 
+/// A common trait for the ability to compute a
+/// cryptographic hash.
+pub trait CryptoHash {
```

Comment:
Tbh, I would like something like `Hashable` more than `*Hash`. `*Hash` looks like a result of hashing, which is strengthened by the fact that we use `Hash` for a result of hashing. OTOH, `Hash` from the standard library denotes something that can be hashed, so I'm not really sure.

Reply:
Trait names in Rust, especially for single-method traits, are verbs: it is `Clone`, `Debug` and not `Clonable` and `Debuggable`. So I am pretty strongly against `able`izms :)

Reply:
Half-serious note: It could be most logical to rename `Hash` to `Digest` / `CryptoDigest` then, as [it's already named](https://dnaq.github.io/sodiumoxide/sodiumoxide/crypto/hash/sha256/struct.Digest.html) in `sodiumoxide`.

Reply:
So that this looks like

```
trait CryptoDigest {
    fn digest(&self) -> Digest;
}
```

? 

Yeah, I think I like these names slightly more than `Hash` ones. The diff will be massive though, but that's OK :-)

Reply:
I meant rather something like

```rust
trait CryptoHash {
    fn hash(&self) -> CryptoDigest;
}
```

(or the variant without `Crypto` prefixes, as per [avoiding stuttering guidelines](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#avoid-redundant-prefixes-[rfc-356])). This would be most close to `sodiumoxide`; it uses [`hash(&[u8]) -> Digest`](https://dnaq.github.io/sodiumoxide/sodiumoxide/crypto/hash/sha256/fn.hash.html).

Reply:
I wonder what was the original reasoning behind renaming `Hash` to `Digest` though?..

Overall, I don't have particularly strong feelings here: any combination of names would do, and the status quo has the obvious advantage of being a status quo.

However, if we do change the names, I would like to change the method's name from `hash` to `digest`, to avoid a clash with `std::hash::Hash::hash`

Reply:
If I understand correctly, the most proper term for the result of hashing is *hash digest*; calling it *hash* is a colloquialism (see [Wikipedia](https://en.wikipedia.org/wiki/Hash_function)). I'm not particularly against staying with `Hash` because of the status quo, even as `Hash` vs `CryptoHash` distinction looks a bit confusing imo.